xrdp/xrdp_mm.c

```diff
@@ -233,85 +220,63 @@ xrdp_mm_get_value_int(struct xrdp_mm *self, const char *aname, int def)
 /*****************************************************************************/
 /* Send gateway login information to sesman */
 static int
-xrdp_mm_send_gateway_login(struct xrdp_mm *self, const char *username,
-                           const char *password)
+xrdp_mm_send_sys_login_request(struct xrdp_mm *self, const char *username,
+                               const char *password)
 {
     xrdp_wm_log_msg(self->wm, LOG_LEVEL_DEBUG,
                     "sending login info to session manager, please wait...");
 
-    return scp_send_gateway_request(
+    return scp_send_sys_login_request(
                self->sesman_trans, username, password,
                self->wm->client_info->client_ip);
 }
 
 /*****************************************************************************/
-/* Send login information to sesman */
+/* Send a create session request to sesman */
 static int
-xrdp_mm_send_login(struct xrdp_mm *self)
+xrdp_mm_create_session(struct xrdp_mm *self)
 {
     int rv = 0;
     int xserverbpp;
-    const char *username;
-    const char *password;
+    enum scp_session_type type;
 
-    username = xrdp_mm_get_value(self, "username");
-    password = xrdp_mm_get_value(self, "password");
-    if (username == NULL || username[0] == '\0')
-    {
-        xrdp_wm_log_msg(self->wm, LOG_LEVEL_ERROR, "No username is available");
-        rv = 1;
-    }
-    else if (password == NULL)
-    {
-        /* Can't find a password definition at all - even an empty one */
-        xrdp_wm_log_msg(self->wm, LOG_LEVEL_ERROR,
-                        "No password field is available");
-        rv = 1;
-    }
-    else
+    /* Map the session code to an SCP session type */
+    switch (self->code)
     {
-        enum scp_session_type type;
-        /* Map the session code to an SCP session type */
-        switch (self->code)
-        {
-            case XVNC_SESSION_CODE:
-                type = SCP_SESSION_TYPE_XVNC;
-                break;
+        case XVNC_SESSION_CODE:
+            type = SCP_SESSION_TYPE_XVNC;
+            break;
 
-            case XRDP_SESSION_CODE:
-                type = SCP_SESSION_TYPE_XRDP;
-                break;
+        case XRDP_SESSION_CODE:
```

Comment:
I think we can remove this support for [x11rdp](https://github.com/neutrinolabs/x11rdp) in xrdp 1.0. It is replaced with xorgxrdp and been kept for compatibility for a while. It's time to remove it because xrdp 1.0 is a big jump.

However, it can be done later.

Reply:
Agreed.